wordpress-mobile/WordPress-Android
# Prepublishing Nudges : Home Screen UI Update

Fixes #11697

## Findings
The skeleton bottom sheet that was created at the start of this project contained the core components required to continue the UI improvements so it was extended to support the header & publish button as seen in the design here. 
<img width="669" alt="Screen Shot 2020-04-28 at 3 43 19 PM" src="https://user-images.githubusercontent.com/1509205/80535833-082c9e00-8967-11ea-9e77-2b0d5851848e.png">

## Solution

In this solution the main changes that were made are 

1. The Header UI was added and it is powered by data from the `SiteModel` associated with the `EditPostRepository`. Within the design above, the site image is rounded, so the `ImageManager` was modified to include functionality that performs rounded corners transformation on images. 

2. The Publish Button was added  5f4adffc4cd38bfc820e43179cdcfb05c863c569  2f7c8c8ccd697a3690f1e2c57fee791f50a91dab  8a999e6bdbc993fd3a3cd211b84c2c500f4c5287 . Currently, it doesn't actually do anything as that will be done in another PR https://github.com/wordpress-mobile/WordPress-Android/issues/11604

3. The `BottomSheet` height was made dynamic since it needs to adapt to several different screen sizes and the UI designs stipulate that it takes up only a portion of the screen real estate so there's logic that calculates the height based on the screen size & orientation. 386a146c4517a8e87cbc0d50f961de17bc9157f0

## Testing

1. Create a new post on press publish. 
2. You should see the bottom sheet. 
3. Ensure that you utilize multiple device types and orientations. 

## Phone 

#### Portrait 

Light Mode | Dark Mode 
--------|-------
<img src="https://user-images.githubusercontent.com/1509205/80536699-627a2e80-8968-11ea-8211-7a7b0bf40bc5.png" width="320">| <img src="https://user-images.githubusercontent.com/1509205/80536696-61490180-8968-11ea-854a-674fdc1a87ec.png" width="320">  

#### Landscape

Light Mode | 
--------| 
![Screenshot_1588107460](https://user-images.githubusercontent.com/1509205/80537153-28f5f300-8969-11ea-9d90-86288d20bc7d.png)
Dark Mode |  
![Screenshot_1588107448](https://user-images.githubusercontent.com/1509205/80537151-27c4c600-8969-11ea-8aff-80f233dc4cdc.png)

## Small Tablet  

### Portrait

Light Mode | Dark Mode 
--------|-------
![Screenshot_1588107873](https://user-images.githubusercontent.com/1509205/80537616-f7c9f280-8969-11ea-9fc5-6153b87add5d.png)| ![Screenshot_1588105842](https://user-images.githubusercontent.com/1509205/80537633-ff899700-8969-11ea-821d-9194773cdaf3.png)

### Landscape

Light Mode | 
--------| 
![Screenshot_1588108109](https://user-images.githubusercontent.com/1509205/80537971-8a6a9180-896a-11ea-9354-1f6c64b4f940.png)
Dark Mode | 
![Screenshot_1588108119](https://user-images.githubusercontent.com/1509205/80537974-8b9bbe80-896a-11ea-830c-e7fa56fc600c.png)

## Large Tablet  

### Portrait

Light Mode | Dark Mode 
--------|-------
![Screenshot_1588108360](https://user-images.githubusercontent.com/1509205/80538311-2399a800-896b-11ea-8238-c04370fb9cc3.png)| ![Screenshot_1588108372](https://user-images.githubusercontent.com/1509205/80538315-24cad500-896b-11ea-9c6c-040dddce7c45.png)

### Landscape

Light Mode | 
--------| 
![Screenshot_1588110686](https://user-images.githubusercontent.com/1509205/80541399-b4bf4d80-8970-11ea-9560-d49c75a8ccf8.png)
Dark Mode | 
![Screenshot_1588110696](https://user-images.githubusercontent.com/1509205/80541402-b6891100-8970-11ea-9aa8-9ab074e2b960.png)

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
